components/localize/localize-mixin.js

```diff
@@ -0,0 +1,166 @@
+import IntlMessageFormat from 'intl-messageformat/src/main.js';
+window.IntlMessageFormat = IntlMessageFormat;
+
+export const D2LLocalizeMixin = superclass => class extends superclass {
```

Comment:
Super nit: I'd be cool with just calling this `LocalizeMixin`, since everything we make in here is "D2L" :)